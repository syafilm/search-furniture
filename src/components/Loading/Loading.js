import React from 'react'
import PropTypes from 'prop-types'

import {
  Styles,
} from 'utils'

const back = 'back'
const front = 'front'
const width = '16px'
const height = '12px'
const point1 = '1 6 4 6 6 11 10 1 12 6 15 6'
const point2 = '1 6 4 6 6 11 10 1 12 6 15 6'

const Loading = ({ color }) => (
  <div className={Styles.indicator(color)}>
    <svg width={width} height={height}>
      <polyline id={back} points={point1} />
      <polyline id={front} points={point2} />
    </svg>
  </div>
)

Loading.propTypes = {
  color: PropTypes.string,
}

Loading.defaultProps = {
  color: '',
}

export default Loading
