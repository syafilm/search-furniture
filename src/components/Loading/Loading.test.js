import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Loading from './Loading'

describe('Component Loading', () => {
  const light = '#fff'
  const dark = '#000'
  it('Can render Loading smoothly with dark mode', () => {
    const component = shallow(
      <Loading color={light} />,
    )
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
    expect(component.find('div')).toBeDefined()
  })

  it('Can render Loading smoothly with light mode', () => {
    const component = shallow(
      <Loading color={dark} />,
    )
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
    expect(component.find('div')).toBeDefined()
  })
})
