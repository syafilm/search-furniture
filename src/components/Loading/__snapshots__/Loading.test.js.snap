// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Loading Can render Loading smoothly with dark mode 1`] = `
<div
  className="css-1pnfice"
>
  <svg
    height="12px"
    width="16px"
  >
    <polyline
      id="back"
      points="1 6 4 6 6 11 10 1 12 6 15 6"
    />
    <polyline
      id="front"
      points="1 6 4 6 6 11 10 1 12 6 15 6"
    />
  </svg>
</div>
`;

exports[`Component Loading Can render Loading smoothly with light mode 1`] = `
<div
  className="css-1xkqot3"
>
  <svg
    height="12px"
    width="16px"
  >
    <polyline
      id="back"
      points="1 6 4 6 6 11 10 1 12 6 15 6"
    />
    <polyline
      id="front"
      points="1 6 4 6 6 11 10 1 12 6 15 6"
    />
  </svg>
</div>
`;
